spacy/cli/assemble.py

```diff
@@ -19,7 +19,7 @@ def assemble_cli(
     ctx: typer.Context,  # This is only used to read additional arguments
     config_path: Path = Arg(..., help="Path to config file", exists=True, allow_dash=True),
     output_path: Path = Arg(..., help="Output directory to store assembled pipeline in"),
-    code_path: Optional[Path] = Opt(None, "--code", "-c", help="Path to Python file with additional code (registered functions) to be imported"),
+    code_paths: str = Opt("", "--code", "-c", help="Comma-separated paths to Python files with additional code (registered functions) to be included in the package"),
```

Comment:
I wonder whether we can make this less breaking for people calling into this from python? We can probably keep the name `code_path` without it being a real headache and also still allow `None` and `Path`?

But I'm not sure what others think?

Reply:
Given that this is for v4, I think it's more important that this argument be consistent between commands than consistent with previous versions.

Reply:
I agree with Adriane that this feels unnecessarily breaking, even for v4. Users will have their CI scripts and it'll be annoying to change all of these if you're only using a single path. It looks like right now, the only command using `code_paths` is `package` and I agree that it's slightly unfortunate not to have this consistent, but I'm not sure the consistency is worth the disruption.

Reply:
OK, I'll go ahead and change it, but to be clear this only affects usage from Python - I expect most people using this feature are using command line scripts, which would be unaffected.

About the types, I think one issue here is that there are no Union types for arguments, so it has to be `str` or `Path`. I am not sure if #12258 will fix that or not.

Reply:
Actually, on thinking about this a little more, I'm not sure it even makes sense to use this argument if you're calling these functions from Python, since you can just import the modules yourself directly in that case. 

That said, I have changed the names back, and I will leave them that way for now.

Reply:
@svlandeg @adrianeboyd As this has been renamed, are we good to go ahead and close this thread?

Reply:
Looking at this again, since we can't support any of the original types for `code_path` it will break existing code anyway, so I would go back to `code_paths` for the new comma-separated option.

To be less breaking, we could continue to support `code_path` as a hidden option / python argument as it was before?

In general it might be nice to support the more frequent comma-hassle-avoiding kind of option where you provide the option multiple times for multiple files like `-c file1.py -c file2.py` but I'm not sure it's really worth this change in this particular case.

Reply:
We use `code_path` for the comma-separated option right now (except in `package`, where we use `code_paths`). Do we want to use `code_paths` for comma-separated values and `code_path` for BW compatibility or are we happy with things as they are?